packages/marketplace/src/core/private-route.tsx

```diff
@@ -40,14 +43,18 @@ export const handleChangeLoginType = (
     if (!loginIdentity || isFetchingAccessToken) {
       return
     }
-    if (loginType === 'CLIENT' && allow === 'DEVELOPER' && loginIdentity.developerId) {
-      dispatch(authChangeLoginType('DEVELOPER'))
+    if (loginType !== 'ADMIN' && allow === 'ADMIN' && loginIdentity.adminId) {
+      dispatch(authChangeLoginType('ADMIN'))
       return
     }
-    if (loginType === 'DEVELOPER' && allow === 'CLIENT' && loginIdentity.clientId) {
+    if (loginType !== 'CLIENT' && allow === 'CLIENT' && loginIdentity.clientId) {
```

Comment:
I think @nphivu414 clear about this one and can answer you what are happening here. This is not the solution for authenticate admin, client, and developer right now. This change will cause another bug

Reply:
Hi @tanphamhaiduong ,
The new logic here is still correct, it also covers the case when users change URL from client/dev portal to admin portal. The only problem is that this logic looks a bit unclear and it's hard for other members to understand what's going on and why we need to do these things.

Reply:
@nphivu414 Thanks for information. @willmcvay  I think this place should change to look more cleaner. Because it's very hard to understand what are happening in here